File: Missed_Data_and_Average.cpp
#include<iostream>

using namespace std;

#include<vector>
#include <iomanip>

int main()

{

vector<int> v;
int input; cin>>input;
while (input!=-1)
{
    v.push_back(input);
    cin>>input;
}


vector<int>::iterator ptr;

int missed_pos, missed_element;
cin>>missed_pos>>missed_element;
v.insert(v.begin()+missed_pos-1,missed_element);

for(ptr = v.begin();ptr < v.end();ptr++)

{

cout<<*ptr<<"\t";

}
cout<<'\n';
for(int index = 0;index < v.size()-2; index++)
{
  double avg = (v[index]+v[index+1]+v[index+2])/3.0;
  cout<<fixed<<setprecision(2)<<avg<<'\t';
}
}

File: Missed Data and Average.py
data =[]
element = int(input())
while element!=-1:
    data.append(element)
    element = int(input())
missing_pos,missing_element = int(input()),int(input())
data.insert(missing_pos-1,missing_element)
for num in data:
    print(num,end='\t')
print()
for i in range(0,len(data)-2):
    avg = sum(data[i:i+3])/3
    print(f'{avg:.2f}',end='\t')
